src/context/providers/HubbardContextProvider.js

```diff
@@ -0,0 +1,83 @@
+import { Application } from "@exabyte-io/ade.js";
+import {
+    ApplicationContextMixin,
+    JSONSchemaFormDataProvider,
+    MaterialContextMixin,
+} from "@exabyte-io/code.js/dist/context";
+import { Made } from "@exabyte-io/made.js";
+import { mix } from "mixwith";
+
+const defaultHubbardConfig = {
+    atomicSpecies: "",
+    atomicOrbital: "2p",
+    hubbardUValue: 0.01,
+};
+
+export class HubbardContextProvider extends mix(JSONSchemaFormDataProvider).with(
+    ApplicationContextMixin,
+    MaterialContextMixin,
+) {
+    static Application = Application;
+
+    static Material = Made.Material;
+
+    constructor(config) {
+        super(config);
+        this.uniqueElements = this.material.Basis.uniqueElements;
+    }
+
+    get defaultData() {
+        return [{ ...defaultHubbardConfig, atomicSpecies: this.uniqueElements }];
+    }
+
+    get uiSchemaStyled() {
+        return {
+            "ui:options": {
+                addable: true,
+                orderable: false,
+                removable: true,
+            },
+            title: {
+                classNames: "col-xs-12",
+            },
+            items: {
+                atomicSpecies: this.defaultFieldStyles,
+                atomicOrbital: this.defaultFieldStyles,
+                hubbardUValue: this.defaultFieldStyles,
+            },
+        };
+    }
+
+    // eslint-disable-next-line class-methods-use-this
+    get jsonSchema() {
+        return {
+            $schema: "http://json-schema.org/draft-04/schema#",
+            title: "",
+            description: "Hubbard parameters for DFT+U calculation.",
+            type: "array",
+            items: {
+                type: "object",
+                properties: {
+                    atomicSpecies: {
+                        type: "string",
+                        title: "Atomic species",
+                        enum: this.uniqueElements,
+                        default: this.uniqueElements[0],
```

Comment:
let's only access the zeroth element if the list is not empty, perhaps something like this:
```javascript
default: this.uniqueElements?.length ? this.uniqueElements[0] : "",
```